lib/oli_web/plugs/redirect_by_attempt_state.ex

```diff
@@ -222,22 +222,46 @@ defmodule OliWeb.Plugs.RedirectByAttemptState do
     end
   end
 
-  defp ensure_path(conn, rendering_type) when rendering_type in [:adaptive_lesson, :lesson] do
+  defp ensure_path(conn, rendering_type) when rendering_type == :lesson do
```

Comment:
Here we could use pattern matching in the definition of the function, WDYT? 
I think it should work anyway by doing something like : 

`defp ensure_path(conn, :lesson) do`

Reply:
Totally agree! I adapted the code and didn't realize the overhead the `when` clause is adding here